src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,109 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
+        int hash = hash(key);
+        int index = (table.length - 1) & hash;
+        Node<K, V> newNode = new Node(hash, key, value, null);
+
+        if (table[index] == null) {
+            table[index] = newNode;
+        } else {
+            Node<K, V> current = table[index];
+            while (current != null) {
+                if (current.hash == hash
+                        && (current.key == key || (key != null && key.equals(current.key)))) {
+                    current.value = value;
+                    return;
+                }
+                if (current.next == null) {
+                    current.next = newNode;
+                    break;
+                }
+                current = current.next;
+            }
+        }
 
+        size++;
+        resizeIfNeeded();
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node<K, V> node = getNode(key);
+        return (node == null) ? null : node.value;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int hash(K key) {
+        return (key == null) ? 0 : Math.abs(key.hashCode());
+    }
+
+    private Node<K, V> getNode(K key) {
+        int hash = hash(key);
+        int index = (table.length - 1) & hash;
+        Node<K, V> node = table[index];
+
+        while (node != null) {
+            if (node.hash == hash
+                    && (node.key == key || (key != null && key.equals(node.key)))) {
+                return node;
+            }
+            node = node.next;
+        }
+        return null;
+    }
+
+    private void resizeIfNeeded() {
+        if (size > threshold) {
```

Comment:
We want resize hash table before size becomes larger than threshold value
```suggestion
        if (size == threshold) {
```